ra2ce/analysis/adaptation/adaptation_option.py

```diff
@@ -148,3 +204,36 @@ def calc_cost(cost: float, year: float) -> float:
                 _lifetime_cost += calc_cost(self.maintenance_cost, _maint_year)
 
         return _lifetime_cost
+
+    def calculate_impact(self, benefit_graph: GeoDataFrame) -> GeoDataFrame:
+        """
+        Calculate the impact of the adaptation option.
+
+        Returns:
+            float: The impact of the adaptation option.
+        """
+        # Damages analysis
+        _damages = Damages(self.damages_input)
+        _damages_gdf = _damages.execute()
+        _dam_col = _damages_gdf.filter(regex="dam_").columns[0]
+        benefit_graph[f"{self.id}_{_dam_col}"] = _damages_gdf[_dam_col]
+
+        # Losses analysis
+        if self.losses_analysis is AnalysisLossesEnum.SINGLE_LINK_LOSSES:
+            _losses = SingleLinkLosses(self.losses_input, self.analysis_config)
+        elif self.losses_analysis is AnalysisLossesEnum.MULTI_LINK_LOSSES:
+            _losses = MultiLinkLosses(self.losses_input, self.analysis_config)
+        else:
+            raise NotImplementedError(
+                f"Losses analysis {self.losses_analysis} not implemented"
+            )
```

Comment:
Encapsulate in a factory (static) method.

Reply:
Done